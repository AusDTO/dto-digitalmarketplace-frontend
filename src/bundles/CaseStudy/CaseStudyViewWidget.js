import React from 'react'
import { Provider } from 'react-redux'
import Registry from '../../registry'

import createStore from './redux/create'
import View from './components/View'

const CaseStudyViewWidget = (props, _context) => {
  const store = createStore(props)
  return (
    <Provider store={store}>
      <View />
    </Provider>
  )
}


Registry.add({
  'casestudy-view': CaseStudyViewWidget
})

export default Registry
