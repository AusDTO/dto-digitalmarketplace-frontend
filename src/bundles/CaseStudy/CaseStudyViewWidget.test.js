// FIXME This is a workaround.
// @see https://github.com/facebook/react/issues/7386
jest.mock('react-dom');

import React from 'react';
import { CaseStudyViewWidget } from './CaseStudyViewWidget';
import renderer from 'react-test-renderer';

test('CaseStudyViewWidget renders without errrors', () => {
  const state = {
    casestudy: {
      title: 'A Case Study',
      opportunity: 'A Case Study Opportunity',
      client: 'A Case study client',
      approach: 'A simple approach',
      timeframe: '2016',
      outcome: ['a', 'b'],
      projectLinks: ['http://facebook.com']
    }
  }
  const component = renderer.create(
    <CaseStudyViewWidget {...state} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})