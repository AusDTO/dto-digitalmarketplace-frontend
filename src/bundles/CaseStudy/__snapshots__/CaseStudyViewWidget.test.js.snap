exports[`test CaseStudyViewWidget renders without errrors 1`] = `
<section>
  <header
    className="row">
    <div
      className="col-md-12">
      <h2>
        A Case Study
      </h2>
    </div>
    <div
      className="meta col-md-12">
      <div
        className="row">
        <div
          className="col-sm-12 col-md-7">
          <small>
            A Case study client
          </small>
        </div>
      </div>
    </div>
  </header>
  <aside
    className="sidebar">
    <h4>
      Timeframe
    </h4>
    <p>
      2016
    </p>
  </aside>
  <article
    className="content-main row"
    role="main">
    <div
      className="col-md-8 col-sm-12">
      <section>
        <h3>
          Objective
        </h3>
        <p>
          A Case Study Opportunity
        </p>
      </section>
      <section>
        <h3>
          Approach
        </h3>
        <p>
          A simple approach
        </p>
      </section>
      <section>
        <h3>
          Outcomes and benefits
        </h3>
        <ul>
          <li>
            a
          </li>
          <li>
            b
          </li>
        </ul>
      </section>
      <section>
        <h3>
          Visit the product
        </h3>
        <a
          href="http://facebook.com"
          rel="external">
          http://facebook.com
        </a>
      </section>
    </div>
  </article>
</section>
`;
